tests/arch_tests.cmake

```diff
@@ -520,6 +520,92 @@ function(add_xc7_validation_test)
     endforeach()
 endfunction()
 
+function(add_xc7_timing_test)
+    # ~~~
+    # add_xc7_timing_test(
+    #    name <name>
+    #    device <device>
+    #    board <board>
+    # )
+    #
+    # Patches xc7 devices with timing data from prjxray-db
+    #
+    # Arguments:
+    #   - name: test name
+    #
+    # Targets generated:
+    #   - vivado_report.txt    : vivado timing report
+    #   - custom_report.txt    : python fpga interchange STA timing report
+    #   - compare_report.txt   : compare report between vivado and python fpga interchange STA values
+
+    set(options)
+    set(oneValueArgs name device board)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        add_xc7_timing_test
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(device ${add_xc7_timing_test_device})
+    set(name ${add_xc7_timing_test_name})
+    set(board ${add_xc7_timing_test_board})
+
+    set(quiet_cmd ${CMAKE_SOURCE_DIR}/utils/quiet_cmd.sh)
+    set(test_name "${name}-${board}")
+    set(dcp ${output_dir}/${name}.dcp)
+    set(run_vivado ${CMAKE_SOURCE_DIR}/utils/run_vivado.sh)
+    set(vivado_report ${output_dir}/vivado_report.txt)
+    set(vivado_timing_tcl ${CMAKE_SOURCE_DIR}/tests/common/timing_dump_vivado.tcl)
+    message(STATUS "${arch} ${test_name} ${board} ${name}")
```

Comment:
This might be a leftover, if we want to keep prints like these I think they should be output only in debug mode.